File: app/src/main/java/com/faendir/lightning_launcher/multitool/event/SwitchFragmentRequest.java
package com.faendir.lightning_launcher.multitool.event;

import androidx.annotation.StringRes;
import com.faendir.lightning_launcher.multitool.util.Fragments;

/**
 * Created on 01.04.2016.
 *
 * @author F43nd1r
 */
public class SwitchFragmentRequest {
    private final Fragments fragment;

    public SwitchFragmentRequest(@StringRes int id) {
        this(Fragments.stream().filter(f -> f.getRes() == id).findAny().orElseThrow());
    }

    public SwitchFragmentRequest(Fragments fragment) {
        this.fragment = fragment;
    }

    public Fragments getFragment() {
        return fragment;
    }

    @StringRes
    public int getId() {
        return fragment.getRes();
    }
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/event/SwitchFragmentRequest.kt
package com.faendir.lightning_launcher.multitool.event

import androidx.annotation.StringRes
import com.faendir.lightning_launcher.multitool.util.Fragments

/**
 * Created on 01.04.2016.
 *
 * @author F43nd1r
 */
class SwitchFragmentRequest(val fragment: Fragments) {

    val id: Int
        @StringRes
        get() = fragment.res

    constructor(@StringRes id: Int) : this(Fragments.values().first { it.res == id })
}
